pytket/pytket/wasm/wasm.py

```diff
@@ -64,169 +72,181 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
+                        _func_lookup[f_name] = (f_idx, entry.index)
+                        f_idx += 1
+
+            # read in map of function signatures to function names
+            elif cur_sec_data.id == SEC_FUNCTION:
+                self._function_types = cur_sec_data.payload.types
+
+        for x in function_names:
+            # check for only integer type in parameters and return values
+            supported_function = True
+            idx = _func_lookup[x][1]
+
+            if idx >= len(self._function_types):
+                raise ValueError("invalid wasm file")
+
+            for t in function_signatures[self._function_types[idx]]["parameter_types"]:
+                if t != self._int_type:
+                    supported_function = False
+            for t in function_signatures[self._function_types[idx]]["return_types"]:
+                if t != self._int_type:
                     supported_function = False
 
-                if supported_function:
-                    self._functions[x] = (
-                        len(
-                            function_signatures[self._function_types[idx]][
-                                "parameter_types"
-                            ]
-                        ),
-                        len(
-                            function_signatures[self._function_types[idx]][
-                                "return_types"
-                            ]
-                        ),
-                    )
+            if len(function_signatures[self._function_types[idx]]["return_types"]) > 1:
+                supported_function = False
+
+            if supported_function:
+                self._functions[x] = (
+                    len(
+                        function_signatures[self._function_types[idx]][
+                            "parameter_types"
+                        ]
+                    ),
+                    len(function_signatures[self._function_types[idx]]["return_types"]),
+                )
+
+            if not supported_function:
+                self._unsupported_functions.append(x)
 
-                if not supported_function:
-                    self._unsupported_function.append(x)
+        if "init" not in self._functions:
+            raise ValueError("wasm file needs to contain a function named 'init'")
 
-            if "init" not in self._functions:
-                raise ValueError("wasm file needs to contain a function named 'init'")
+        if self._functions["init"][0] != 0:
+            raise ValueError("init function should not have any parameter")
 
-            if self._functions["init"][0] != 0:
-                raise ValueError("init function should not have any parameter")
+        if self._functions["init"][1] != 0:
+            raise ValueError("init function should not have any results")
 
-            if self._functions["init"][1] != 0:
-                raise ValueError("init function should not have any results")
+        self.checked = True
```

Comment:
I probably should include a comment about the importance of this line

Reply:
Yes would be good